Python Examples/TestMonitor/CreateResultsAndSteps/create_results_and_steps.py

```diff
@@ -0,0 +1,310 @@
+    :param voltage: The electrical voltage value.
+    :return: A tuple containing the electrical power measurements and the input and output lists.
+    """
+    current_loss = 1 - random.uniform(0, 1) * 0.25
+    voltage_loss = 1 - random.uniform(0, 1) * 0.25
+    power = current * current_loss * voltage * voltage_loss
+
+    # Record electrical current and voltage as inputs.
+    inputs = [{"name":"current", "value":current}, {"name":"voltage", "value":voltage}]
+
+    # Record electrical power as an output.
+    outputs = [{"name":"power", "value":power}]
+
+    return power, inputs, outputs
+
+
+def build_power_measurement_params(power: float, low_limit: float, high_limit: float, status: Dict) -> Dict:
+    """
+    Builds a Test Monitor measurement parameter object for the power test.
+    :param power: The electrical power measurement.
+    :param low_limit: The value of the low limit for the test.
+    :param high_limit: The value of the high limit for the test.
+    :param status: The measurement's pass/fail status.
+    :return: A list of test measurement parameters.
+    """
+    parameter = {
+        "name": "Power Test",
+        "status": str(status["statusType"]),
+        "measurement": str(power),
+        "units": "Watts",
+        "nominalValue": None,
+        "lowLimit": str(low_limit),
+        "highLimit": str(high_limit),
+        "comparisonType": "GELE"
+    }
+
+    parameters = {"text": "", "parameters": [parameter]}
+    return parameters
+
+
+def remove_if_key_exists(dict: Dict, key: str) -> None:
+    if key in dict.keys():
+        dict.pop(key)
+
+
+def generate_step_data(
+    name: str,
+    step_type: str,
+    inputs: List[Dict] = None,
+    outputs: List[Dict] = None,
+    parameters: Dict = None,
+    status: Dict = None,
+) -> Dict:
+    """
+    Creates the step data and
+    populates it to match the TestStand data model.
+    :param name: The test step's name.
+    :param step_type: The test step's type.
+    :param inputs: The test step's input values.
+    :param outputs: The test step's output values.
+    :param parameters: The measurement parameters.
+    :param status:
+    :return: The step data used to create a test step.
+    """
+    step_status = status if status else {
+        "statusType": "RUNNING",
+        "statusName": "Running"
+        }
+
+    step_data = {
+        "stepId": None,
+        "parentId": None,
+        "resultId": None,
+        "children": None,
+        "data": parameters,
+        "dataModel": "TestStand",
+        "name": name,
+        "startedAt":  str(datetime.datetime.utcnow()),
+        "status": step_status,
+        "stepType": step_type,
+        "totalTimeInSeconds": random.uniform(0, 1) * 10,
+        "inputs": inputs,
+        "outputs": outputs
+    }
+
+    return step_data
+
+
+def is_partial_success_response(response: Dict) -> bool:
+    return "error" in response.keys()
+
+
+def get_test_result() -> Dict:
```

Comment:
It's strange to me that this is called `get...` beacuse we're not actually getting anything (from the server). I would expect this to be something more like `create_test_result` or something. Also, I think maybe this belongs in the client. In a real implementation, I would expect to have a client method that takes parameters for all the elements of a result and returns a well formatted request.

Reply:
Created a `create_test_result` method in the client, which takes all elements of the result as input parameters and returns a formatted result request.